R/financial_functions.R

```diff
@@ -45,23 +45,75 @@ calculate_net_profits <- function(data,
     )
 }
 
+#' Calculates annual net profits on the company-technology level for the
+#' baseline and late and sudden scenarios without carbon tax Climate laggards which need to build
+#' out their production in increasing technologies to compensate for their
+#' missed targets, are "punished" by adjusting the net profit margin on their
+#' additional build out based on their proximity to target within the given
+#' technology. Specifically, we measure the ratio of how much of the required
+#' build out or reduction in a technology the company will have done at the end
+#' of the forecast period. If the technology has an increasing target and the
+#' ratio of completion is below one, the net_profit_margin on the additional
+#' production build out is multiplied with the proximity to the target. This
+#' approximates the additional capital investment such a company would have to
+#' make in a short time, which leads to added costs. This ensures that late
+#' build out will not proportionally translate into increased profits.
+#'
+#' @param data A data frame containing the production forecasts of companies
+#'   under baseline and late and sudden, market prices/costs, company net profit
+#'   margins, the proximity to target in the production forecast period and an
+#'   indication of the direction of the technology.
+#'   @param direction either declining or increasing
+
 calculate_net_profits_without_carbon_tax <- function(data) {
+  baseline <- calculate_net_profits_baseline(data)
+  shock_increasing_technologies <- calculate_net_profits_shock_increasing_technologies(data = data %>% dplyr::filter(direction == "increasing"))
+  shock_declining_technologies <- calculate_net_profits_shock_declining_technologies(data = data %>% dplyr::filter(direction == "declining"))
+
+  data_test <- dplyr::full_join(shock_increasing_technologies, shock_declining_technologies)
+  data_test <- dplyr::full_join(data_test, baseline)
+}
+
+#' Calculates annual net profits on the company-technology level for the
+#' baseline
+#' @param data A data frame containing the production forecasts of companies
+#'   under baseline and late and sudden, market prices/costs, company net profit
+#'   margins, the proximity to target in the production forecast period and an
+#'   indication of the direction of the technology.
+
+calculate_net_profits_baseline <- function(data) {
+  data <- data %>%
```

Comment:
I would suggest to explicitly return from functions, i.e. put return(data) at the end. It is not always needed, but there is some rare cases where not returning explicitly will return the wrong thing..

Reply:
done